realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -428,6 +418,11 @@ private void enableCollectionSnapshot() {
     }
 
     void disableCollectionSnapshot() {
+        if (isInTransaction()) {
+            // This should never happen.
+            throw new IllegalStateException( "Collection cannot be reattached if the Realm is in transaction." +
+                    " Please remember to commit or cancel transaction before finishing the current event loop.");
```

Comment:
Does this mean you cannot have a transaction that takes longer than 1 event loop? (for example, open a transaction, make object shown editable, then on save do `commitTransaction()`)